### DISCUSSION TITLE
Embedded material and texture from png file

### DISCUSSION THREAD

**Discussion by Pankaj003** [assimp/assimp#4139] [Q&A] [answered]
Hi,
I've  a requirement to create embedded material for gltf output.
I've png files specifying diffuse, occlusion, specular and other properties.
I want to create texture data from these png files. I am not sure how to create the aiTexture structure, specially 
**aiTexel\* pcData** variable.
Could anyone please provide some input on how to achieve this.
Pointing to sample source code will be great.

Thanks,
Pankaj

**Comment by Pankaj003** [comment]
Hi @kimkulling,

Are embedded texture supported for gltf2 export format?
If yes then could you please tell me how to implement that.

Thanks,
Pankaj

**Comment by kimkulling** [comment, marked as answer]
I guess at this moment this feature is not implemented. If you want to deliver a PR for that take a look onto the exporter for 3MF.
